src/mesh/mesh_refinement.cpp

```diff
@@ -117,10 +117,10 @@ void MeshRefinement::RestrictFieldX1(const ParArrayND<Real> &fine,
         for (int ci = csi; ci <= cei; ci++) {
           int i =
               (ci - pmb->c_cellbounds.is(interior)) * 2 + pmb->cellbounds.is(interior);
-          const Real area00 = coords.Area(X1DIR, k, j, i);
-          const Real area01 = coords.Area(X1DIR, k, j + 1, i);
-          const Real area10 = coords.Area(X1DIR, k + 1, j, i);
-          const Real area11 = coords.Area(X1DIR, k + 1, j + 1, i);
+          const Real area00 = coords.faceArea(X1DIR, k, j, i);
+          const Real area01 = coords.faceArea(X1DIR, k, j + 1, i);
+          const Real area10 = coords.faceArea(X1DIR, k + 1, j, i);
+          const Real area11 = coords.faceArea(X1DIR, k + 1, j + 1, i);
```

Comment:
Can this & below be `FaceArea<XNDIR>()`?  If not, is that something we can fix with constexpr?

Reply:
Similarly, is it worth standardizing all N->XNDIR using this PR, maybe with additional XNFACE variables?

Reply:
Agreed it's verbose and unwieldy to use the macros. I just kept forgetting what Xf<1, 1> was referring to.  Especially in terms of the argument order, since the convention switches by variable.  Xf<X1FACE, X1DIR> or Dxf<X1DIR, X1FACE> is at least totally explicit. (I absolutely mixed up both those orders the first time).

It also wouldn't add any type checks, true.  Which would make it hard to enforce, since using the macros would be purely a matter of convention and require much more typing.

...but that's given me a better (worse) idea: maybe we *should* type-check face locations vs edge locations vs directions?  I mean,  Xf<FaceLocation f, Direction d> seems like even more overkill, but at least then we get the compile-time check benefits and it's easy to enforce a readable standard.

Maybe the answer for now is just to use the Xf<int> shortcut everywhere?  That's unambiguous, resembles the old Athena convention, and it's all we need for UniformCartesian, right?  And, it and leaves room for a future typed solution, since changing the full macro only involves changing the shortcut and not a bunch of calls scattered around the code.

Reply:
The `Xf<int,int>` template is so that we can use `Xf` for all references to locations, replacing the `x*f` and `x*s*` functions in Athena++.

That said, we're currently only using `Xf<x,x>` in Parthenon (and Athena++) although that will change soonish when we add the edge-corrections for face centered fields on AMR blocks. We can add an overloaded template `Xf<int>` when both the face and coordinate direction are the same.

For enforcing typing, we would have to change `CoordinateDirection` from an `enum` to `enum class` I believe. Hopefully that wouldn't break anything. Anywhere we really need to covert from an int to this enum would be slightly more awkward. I would be fine with that change as long as people are onboard with the code bloat (in terms of characters added) that would create.

Reply:
I dunno if I want to move `CoordinateDirection` to an `enum class`... that's going to change everything everywhere.

Reply:
I definitely get why the <int, int> template is necessary.  I was proposing something like:
```cpp
template <int dirface, class... Args>
KOKKOS_FORCEINLINE_FUNCTION Real Dxf(Args... args) const {
  Dxf<dirface, dirface>(args);
}
```

i.e., just overloading and calling through.  I misread the diff and thought that a similar overload was already implemented.  I *think* this would be simple enough to do in a general way, even if the specific code above doesn't work, but the difference is obviously minimal.  If this isn't uniformly agreed to be more readable it's not worth it.

Re: making directions (and faces) an enum class, I am abstractly very in favor of this.  Probably not in this PR though; it was just an idea for someday, and as Jonah says it will mean a lot of lines changed, even if there's always a clear way to translate the code.

Reply:
> ```c++
> template <int dirface, class... Args>
> KOKKOS_FORCEINLINE_FUNCTION Real Dxf(Args... args) const {
>   return Dxf<dirface, dirface>(args);
> }
> ```
I see, that should be implemented, I've now added it in. 


> I dunno if I want to move `CoordinateDirection` to an `enum class`... that's going to change everything everywhere.

That's mostly why I'm hesitant to make that change.  But with just the enum, we don't have any type checking, so using the enum everywhere at the moment won't add much. In the few places where `Xf<int,int>` will be used, it could be advantageous to use different enums to differentiate between coordinate direction and face direction.

> Re: making directions (and faces) an enum class, I am abstractly very in favor of this. Probably not in this PR though; it was just an idea for someday, and as Jonah says it will mean a lot of lines changed, even if there's always a clear way to translate the code.

Yep, let's save it for another PR